### ISSUE TITLE
Run via jenkins complains about autoload module layout

### ISSUE THREAD

**Issue by cdenneen** [rodjek/puppet-lint#361]
I can't seem to reproduce this locally but it happens via Jenkins.

I've installed same rvm version locally but still no issues linting.

-bash-4.1$ bundle exec rake lint
manifests/exampleservice.pp - ERROR: exampleservice::exampleservice not in autoload module layout on line 19
manifests/init.pp - ERROR: exampleservice not in autoload module layout on line 19
rake aborted!

/var/lib/jenkins/.rvm/gems/ruby-2.0.0-p481/gems/puppet-lint-0.3.2/lib/puppet-lint/tasks/puppet-lint.rb:25:in `block (2 levels) in initialize'
/var/lib/jenkins/.rvm/gems/ruby-2.0.0-p481/gems/puppet-lint-0.3.2/lib/puppet-lint/tasks/puppet-lint.rb:13:in `block in initialize'
/var/lib/jenkins/.rvm/gems/ruby-2.0.0-p481/bin/ruby_executable_hooks:15:in `eval'
/var/lib/jenkins/.rvm/gems/ruby-2.0.0-p481/bin/ruby_executable_hooks:15:in `<main>'
Tasks: TOP => lint
(See full trace by running task with --trace)

Using this repo to test:
https://github.com/myoung34/puppet-exampleservice

I end up having to add the following to Rakefile in order to get Jenkins to successfully build:

```PuppetLint.configuration.send('disable_autoloader_layout')```

**Comment by rnelson0** [comment]
@cdenneen Is this still an issue for you? If so, can you update to 2.0.0 and see if that resolves it? Thank you.

**Comment by drewhemm** [comment]
I'm getting this issue with Atlassian Bamboo, when performing the following steps:

1. Bamboo checks out a repo containing an R10K Puppetfile
2. R10K is run to checkout the modules in the Puppetfile (into a 'modules' directory)
3 I then run puppet-lint from the modules/MODULENAME directory. This is the code I have in my Rakefile:

sh "puppet-lint manifests --error-level error --no-2sp_soft_tabs-check"

When I SSH into the Bamboo box and cd into the set working directory and run puppet-lint, it completes without errors. Looks like something strange is going on with the path recognition...

**Comment by rnelson0** [comment]
@drewhemm When you run the test interactively, are you just running `puppet-lint manifests --error-level error --no-2sp_soft_tabs-check` as your user, or are you using sudo/su to run `sh "puppet-lint manifests --error-level error --no-2sp_soft_tabs-check"` as the bamboo service account?

I don't have a working Jenkins or Bamboo installation to test with myself.

**Comment by drewhemm** [comment]
Seems I needed to add the --relative flag for it to work.

**Comment by rnelson0** [comment]
@drewhemm your Jenkins test step is now:

sh "puppet-lint manifests --relative --error-level error --no-2sp_soft_tabs-check"

And it works properly? Can we close this now?

**Comment by drewhemm** [comment]
Confirmed. I've attached my Rakefile which is now working from Bamboo after adding --relative. 

[Rakefile.txt](https://github.com/rodjek/puppet-lint/files/396842/Rakefile.txt)

I imagine the same solution applies to Jenkins

**Comment by rnelson0** [comment]
Awesome, thanks so much for the update Andrew!

@cdenneen I'm closing this out but if you can confirm at your leisure that you were able to resolve your issue with Jenkins that would be great. Thanks!

**Comment by cdenneen** [comment]
@rnelson0 puppet-lint 0.3.2 this works fine without any special change.
puppet-lint 2.0.0 still has this error by default.

Creating a custom Rake task (most modules will use rspec helpers to have default rake tasks), probably isn't really a "fix" (Does work in a custom scenario):

```
  modules.each do |m|
    Dir.chdir("modules/" + m) do
      output = `puppet-lint manifests --error-level error --no-2sp_soft_tabs-check --relative`
      if !$?.success?
        puts output
        exit 1
      end
```

However the relative fix is configurable and does make v2.0.0 work so just want to explain here how that works if someone else stumbles upon this post:

```
  PuppetLint.configuration.relative = true
```
or something like
```
PuppetLint::RakeTask.new :lint do |config|
...
  # Compare module layout relative to the module root
  config.relative = true
...
end
```

Thanks @drewhemm for discovering this and @rnelson0 for the follow-ups.

**Comment by ghoneycutt** [comment]
This is the default when using `puppet module generate`

https://github.com/puppetlabs/puppet/blob/master/lib/puppet/module_tool/skeleton/templates/generator/Rakefile#L11
